### PR TITLE
[1.12.x] ca: move ConnectCA.Sign authorization logic to CAManager

### PR DESCRIPTION
**Pull request by rboyer**
Backport of #12609 to 1.12.x

Conflicts:
- agent/consul/connect_ca_endpoint.go

### CODE REVIEW

**Review by freddygv — APPROVED** [review]
LGTM assuming CI does OK

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/631089.

**Pull request merged**
